packages/@sfx/dom-events-plugin/src/dom-events-plugin.ts

```diff
@@ -0,0 +1,140 @@
+import { Plugin, PluginRegistry, PluginMetadata } from '@sfx/core';
+/**
+ * The browser version of the events plugin designed for the SF-X product.
+ * This plugin is responsible for exposing methods that will allow other
+ * plugins to register, unregister, and dispatch events.
+ */
+export default class DomEventsPlugin implements Plugin {
+  get metadata(): PluginMetadata {
+    return {
+      name: 'dom_events_plugin',
+      depends: [],
+    };
+  }
+
+  /**
+   * The core property is meant to hold a reference to the plugin registry
+   * that is passed in during the plugin registration lifecycle event.
+   */
+  core: PluginRegistry;
+  /**
+   * The exposedValue property is the value that the Events Browser Plugin
+   * exposes to the Core entity.
+   */
+  exposedValue: DomEventsPluginExposedValue;
+  /**
+   * The options property is the set of configuration values used to set
+   * up the plugin during construction.
+   */
+  options: DomEventsPluginOptions = {
+    window: typeof window !== "undefined" ? window : undefined,
+    CustomEvent: typeof CustomEvent !== "undefined" ? CustomEvent : undefined,
```

Comment:
I'm not clear on where the `CustomEvent` value (second and third ones on this line) come from. Is it built into TypeScript?